jujugui/static/gui/src/app/app.js

```diff
@@ -798,22 +798,36 @@ YUI.add('juju-gui', function(Y) {
       element.
 
       @method _renderAddedServices
-      @param {Array} services Array of service models.
+      @param {String} hoveredId An id for a service.
     */
-    _renderAddedServices: function(services) {
+    _renderAddedServices: function(hoveredId) {
       var utils = views.utils;
       var db = this.db;
+      var topo = this.views.environment.instance.topo;
+      var ServiceModule = topo.modules.ServiceModule;
+      // Set up a handler for syncing the service token hover. This needs to be
+      // attached only when the component is visible.
+      if (this.hoverService) {
+        this.hoverService.detach();
+      }
+      this.hoverService = topo.on('hoverService', function(service) {
+        this._renderAddedServices(service.id);
+      }, this);
+      // Deselect the active service token.
+      ServiceModule.deselectNodes();
```

Comment:
When a user navigates to the service details the service token will highlight, then when they close the service details `ServiceModule.deselectNodes();` makes the service token deactivate again.